packages/web-components/fast-element/src/observation/arrays.ts

```diff
@@ -679,8 +698,24 @@ let defaultSpliceStrategy: SpliceStrategy = Object.freeze({
         sort: typeof Array.prototype.sort,
         args: any[]
     ): any[] {
+        const map = new Map();
+
+        for (let i = 0, ii = (array || []).length; i < ii; ++i) {
+            map.set(array[i], i);
+        }
+
         const result = sort.apply(array, args);
-        observer.reset(array);
+
+        (array as any).sorted++;
+
+        const sortedItems: number[] = [];
+        for (let i = 0, ii = (array || []).length; i < ii; ++i) {
+            const newIndex = map.get(array[i]);
```

Comment:
This could cause issues if there are duplicates, we should delete the item in the map after we get it.